mmv1/products/dialogflowcx/SecuritySettings.yaml

```diff
@@ -0,0 +1,161 @@
+    required: true
+  - !ruby/object:Api::Type::Enum
+    name: 'redactionStrategy'
+    description: |
+      Defines how we redact data.
+      * REDACT_WITH_SERVICE: Call redaction service to clean up the data to be persisted.
+    values:
+      - :REDACT_WITH_SERVICE
+  - !ruby/object:Api::Type::Enum
+    name: 'redactionScope'
+    description: |
+      Defines what types of data to redact.
+      * REDACT_DISK_STORAGE: On data to be written to disk or similar devices that are capable of holding data even if power is disconnected. This includes data that are temporarily saved on disk.
+    values:
+      - :REDACT_DISK_STORAGE
+  - !ruby/object:Api::Type::String
+    name: 'inspectTemplate'
+    description: |
+      [DLP](https://cloud.google.com/dlp/docs) inspect template name. Use this template to define inspect base settings. If empty, we use the default DLP inspect config.
+      Note: inspectTemplate must be located in the same region as the SecuritySettings.
+      Format: projects/<Project ID>/locations/<Location ID>/inspectTemplates/<Template ID> OR organizations/<Organization ID>/locations/<Location ID>/inspectTemplates/<Template ID>
+  - !ruby/object:Api::Type::String
+    name: 'deidentifyTemplate'
+    description: |
+      [DLP](https://cloud.google.com/dlp/docs) deidentify template name. Use this template to define de-identification configuration for the content. If empty, Dialogflow replaces sensitive info with [redacted] text.
+      Note: deidentifyTemplate must be located in the same region as the SecuritySettings.
+      Format: projects/<Project ID>/locations/<Location ID>/deidentifyTemplates/<Template ID> OR organizations/<Organization ID>/locations/<Location ID>/deidentifyTemplates/<Template ID>
+  - !ruby/object:Api::Type::Array
+    name: 'purgeDataTypes'
+    description: |
+      List of types of data to remove when retention settings triggers purge.
+    item_type: !ruby/object:Api::Type::Enum
+      name: 'purgeDataType'
+      description: |
+        Type of data we purge after retention settings triggers purge.
+        * DIALOGFLOW_HISTORY: Dialogflow history. This does not include Cloud logging, which is owned by the user - not Dialogflow.
+      values:
+        - :DIALOGFLOW_HISTORY
+  - !ruby/object:Api::Type::NestedObject
+    name: 'audioExportSettings'
+    description: |
+      Controls audio export settings for post-conversation analytics when ingesting audio to conversations.
+      If retention_strategy is set to REMOVE_AFTER_CONVERSATION or gcs_bucket is empty, audio export is disabled.
+      If audio export is enabled, audio is recorded and saved to gcs_bucket, subject to retention policy of gcs_bucket.
+      This setting won't effect audio input for implicit sessions via [Sessions.DetectIntent](https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/projects.locations.agents.sessions/detectIntent#google.cloud.dialogflow.cx.v3.Sessions.DetectIntent).
+    properties:
```

Comment:
Should the subfields be required here? In particular `gcsBucket` seems likely.

Reply:
In this case I'm going to say "I don't know". IDK how this interacts with the Agent-level [advancedSettings.audioExportGcsDestination.uri](https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/AdvancedSettings#GcsDestination), and the API is perfectly happy to accept a `SecuritySettings` object without a `gcsBucket` set -> I think it should be left as-is.

(good question though!)